connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -0,0 +1,156 @@
+import { ModelId } from "@dust-tt/types";
+import { CheerioCrawler, Configuration } from "crawlee";
+import turndown from "turndown";
+
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertToDatasource } from "@connectors/lib/data_sources";
+import { Connector } from "@connectors/lib/models";
+import {
+  WebCrawlerConfiguration,
+  WebCrawlerFolder,
+} from "@connectors/lib/models/webcrawler";
+import {
+  reportInitialSyncProgress,
+  syncSucceeded,
+} from "@connectors/lib/sync_status";
+
+export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found.`);
+  }
+  const webCrawlerConfig = await WebCrawlerConfiguration.findOne({
+    where: {
+      connectorId,
+    },
+  });
+  if (!webCrawlerConfig) {
+    throw new Error(`Webcrawler configuration not found for connector.`);
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  let pageCount = 0;
+  let errorCount = 0;
+  const createdFolders = new Set<string>();
+
+  const crawler = new CheerioCrawler(
+    {
+      maxRequestsPerCrawl: 300,
+      maxConcurrency: 10,
+
+      async requestHandler({ $, request, enqueueLinks }) {
+        const extracted = new turndown()
+          .remove(["style", "script", "iframe"])
+          .turndown($.html());
+
+        const pageTitle = $("title").text();
+
+        await enqueueLinks();
+
+        const folders = getAllFoldersForUrl(request.url);
+        for (const folder of folders) {
+          if (!folder.startsWith(webCrawlerConfig.url)) {
+            continue;
```

Comment:
We don't want to "enqueueLinks" if we escaped the domain do we?

Reply:
EnqueueLink() by default stays on the same domain. The settings for staying under the same "folder" did not work as expected so I removed it last minute, I need to write a few more lines of code to add it back. I am planning on doing a PR just for that.